### PR TITLE
Accessmod: Order projects by last activity date AC-94

### PR DESCRIPTION
**Pull request by qgerome**

### CODE REVIEW

File: hexa/core/graphql/__init__.py [hunk 1]
```diff
@@ -28,3 +28,10 @@ def result_page(queryset, page, per_page=None):
         "total_items": paginator.count,
         "items": paginator.page(page),
     }
+
+
+def order_queryset_by(queryset, field_name: str, direction: str):
```
Comment (pvanliefland) [inline hexa/core/graphql/__init__.py:33]: I think we can skip this - see below

File: hexa/plugins/connector_accessmod/schema.py [hunk 2]
```diff
@@ -107,6 +116,12 @@ def resolve_accessmod_projects(
     if teams is not None and len(teams) > 0:
         queryset = queryset.filter(projectpermission__team__id__in=teams)
 
+    order_by = kwargs.get("orderBy", None)
+    if order_by is not None:
+        queryset = order_queryset_by(
```
Comment (pvanliefland) [inline hexa/plugins/connector_accessmod/schema.py:121]: `queryset = queryset.order_by(kwargs["order_by"])`

File: hexa/plugins/connector_accessmod/schema.py [hunk 3]
```diff
@@ -42,6 +43,14 @@
 # Projects
 project_object = ObjectType("AccessmodProject")
 
+project_order_by = EnumType(
+    "AccessmodProjectOrderField",
+    {
+        "UPDATED_AT": "updated_at",
```
Comment (pvanliefland) [inline hexa/plugins/connector_accessmod/schema.py:49]:
```
{
    "NAME_ASC": "name",
    "NAME_DESC": "-updated_at",
    "UPDATED_AT_ASC": "updated_at",
    "UPDATED_AT_DESC": "-updated_at",
}
```

File: hexa/plugins/connector_accessmod/graphql/schema.graphql [hunk 4]
```diff
@@ -132,9 +132,19 @@ enum DeleteAccessmodProjectPermissionError {
     PERMISSION_DENIED
     NOT_FOUND
 }
+
+enum AccessmodProjectOrderField {
```
Comment (pvanliefland) [inline hexa/plugins/connector_accessmod/graphql/schema.graphql:136]: `enum AccessModProjectOrder`

File: hexa/plugins/connector_accessmod/graphql/schema.graphql [hunk 5]
```diff
@@ -132,9 +132,19 @@ enum DeleteAccessmodProjectPermissionError {
...
+    UPDATED_AT
+    NAME
+}
+input AccessmodProjectOrder {
```
Comment (pvanliefland) [inline hexa/plugins/connector_accessmod/graphql/schema.graphql:140]: Not needed anymore

File: config/graphql/schema.graphql [hunk 6]
```diff
@@ -1,5 +1,10 @@
 scalar Date
 scalar DateTime
 
+enum OrderDirection {
```
Comment (pvanliefland) [inline config/graphql/schema.graphql:4]: We can skip this - see below

File: hexa/plugins/connector_accessmod/schema.py [hunk 7]
```diff
@@ -42,6 +43,14 @@
 # Projects
 project_object = ObjectType("AccessmodProject")
 
+project_order_by = EnumType(
+    "AccessmodProjectOrderField",
```
Comment (pvanliefland) [inline hexa/plugins/connector_accessmod/schema.py:47]: `AccessmodProjectOrder`

**Review by pvanliefland — CHANGES_REQUESTED** [review]
I would like to simplify this and turn `order` query params straight into an enum that maps constant values (such as `NAME_DESC`) to Django order params (here, `-name).

This would allow us to skip `OrderDirection`, the compound order field, the `order_queryset_by` method, etc...

**Comment by qgerome** [comment]
Changes applied.

**Pull request merged**